src/js/api/core-loader.js

```diff
@@ -0,0 +1,203 @@
+import ApiQueueDecorator from './api-queue';
+import SimpleModel from '../model/simplemodel';
+import Timer from 'api/timer';
+import Events from 'utils/backbone.events';
+
+let controllerPromise = null;
+
+function loadController() {
+    if (!controllerPromise) {
+        controllerPromise = new Promise(function (resolve) {
+            require.ensure(['controller/controller'], function (require) {
+                const CoreMixin = require('controller/controller');
+                resolve(CoreMixin);
+            }, 'jwplayer.core');
+        });
+    }
+    return controllerPromise;
+}
+
+const CoreModel = function() {};
+Object.assign(CoreModel.prototype, SimpleModel);
+
+const CoreLoader = function CoreSetup(originalContainer) {
+    loadController();
+    this._events = {};
+    this.controller = null;
+    this.model = new CoreModel();
+    this.model._qoeItem = new Timer();
+    this.originalContainer = originalContainer;
+    this.apiQueue = new ApiQueueDecorator(this, () => true);
+};
+
+/* eslint no-unused-vars: 0 */
+
+Object.assign(CoreLoader.prototype, {
+    on: Events.on,
+    once: Events.once,
+    off: Events.off,
+    trigger: Events.trigger,
+    init(options, api) {
+        loadController().then(CoreMixin => {
+            if (!this.apiQueue) {
+                // Exit if `playerDestroy` was called on CoreLoader clearing the config
+                return;
+            }
+            // copy queued commands
+            const commandQueue = this.apiQueue.queue.slice(0);
+            this.apiQueue.destroy();
+            // Assign CoreMixin.prototype (formerly controller) properties to this instance making api.core the controller
+            Object.assign(this, CoreMixin.prototype);
+            this.setup(api, options, this.originalContainer, this._events, commandQueue);
+        });
+    },
+    playerDestroy() {
+        // TODO: cancel async setup
+        if (this.apiQueue) {
+            this.apiQueue.destroy();
+        }
+        this._events =
+            this.apiQueue =
+            this.originalContainer =
+            this.model =
+            this.controller = null;
+    },
+    getContainer() {
+        return this.originalContainer;
+    },
+
+    // These methods read from the model
+    get(property) {
```

Comment:
The model should be passed into setup so properties are set after ready. Can we also make this simpler so that new API methods don't have to be duplicated here?